kafka-commons/src/test/java/uk/gov/homeoffice/digital/sas/transactionsync/KafkaDbTransactionSynchronizerIntegrationTest.java

```diff
@@ -56,13 +60,15 @@ void givenValidRequest_whenSendingCreateRequest_thenTransactionSyncLogsSuccessMe
 
     profileRepository.saveAndFlush(profile);
 
+    messageKey = profile.resolveMessageKey();
```

Comment:
assuming the entityID  gen didn't work, I **think** this may still pass as it might just append a null id to the key.
might be worth asserting the profile id is not null before the log assertion.

Reply:
yep, as discussed not an issue now, resolving.